### PR TITLE
Storage get count

### PR DESCRIPTION
**Pull request by Viiic98**
Create get and count method
Add tests to every one of them

### CODE REVIEW

File: models/engine/db_storage.py [hunk 1]
```diff
@@ -75,3 +75,21 @@ def reload(self):
     def close(self):
         """call remove() method on the private session attribute"""
         self.__session.remove()
+
```
Comment (danrocus1994) [inline models/engine/db_storage.py:78]: should check about when cls is None should return None inmediatly

File: models/engine/db_storage.py [hunk 2]
```diff
@@ -75,3 +75,21 @@ def reload(self):
...
+    def get(self, cls, id):
+        """ Returns the object based on the class name and its ID,
+            or None if not found """
+        objs = self.all(cls)
+        for obj in objs.values():
+            if obj.id == id:
+                return obj
+        return None
+
```
Comment (Danucas) [inline models/engine/db_storage.py:87]: should check about when cls is None should return None inmediatly

File: models/engine/db_storage.py [hunk 3]
```diff
@@ -75,3 +75,21 @@ def reload(self):
...
+    def count(self, cls=None):
+        """ Returns the number of objects in storage matching
+            the given class name. """
+        if cls is not None:
+            n_obj = len(self.all(cls).keys())
+        else:
+            n_obj = len(self.all().keys())
+        return n_obj
```
Comment (Danucas) [inline models/engine/db_storage.py:95]: You could avoid the else statement just return `len(self.all().keys())`

File: models/engine/file_storage.py [hunk 4]
```diff
@@ -68,3 +68,21 @@ def delete(self, obj=None):
...
+        return None
+
+    def count(self, cls=None):
+        """ Returns the number of objects in storage matching
+            the given class name. """
+        if cls is not None:
+            n_obj = len(self.all(cls).keys())
+        else:
+            n_obj = len(self.all().keys())
+        return n_obj
```
Comment (Danucas) [inline models/engine/file_storage.py:88]: can be reduced as the last example

Comment (Danucas) [reply models/engine/file_storage.py:88]: `return len(self.all().keys())`

File: tests/test_models/test_engine/test_file_storage.py [hunk 5]
```diff
@@ -113,3 +113,53 @@ def test_save(self):
...
+        creates a new object of any class
+        and checks if the object saves corectly
+        """
+        state = State()
+        state.name = "Fictional state"
+        storage = FileStorage()
+        storage.new(state)
+        storage.save()
+        self.assertEqual(storage.get(State, state.id), state)
+
```
Comment (Danucas) [inline tests/test_models/test_engine/test_file_storage.py:130]: Can take in account you can also make a chained query and track the performance to the deep

File: tests/test_models/test_engine/test_file_storage.py [hunk 6]
```diff
@@ -113,3 +113,53 @@ def test_save(self):
...
+        """
+        Test the count() function when cls is given
+        """
+        state = State()
+        state.name = "Fictional state"
+        storage = FileStorage()
+        storage.new(state)
+        storage.save()
+        self.assertTrue(storage.count(State) > 0)
+
```
Comment (Danucas) [inline tests/test_models/test_engine/test_file_storage.py:150]: can also take in account incremental an decremental behavior

**Pull request merged**